lib/deref.js

```diff
@@ -337,7 +337,7 @@ module.exports = {
 
       // Discard format if not supported by both json-schema-faker and ajv or pattern is also defined
       if (!_.includes(SUPPORTED_FORMATS, schema.format) || (schema.pattern && schema.format)) {
-        delete schema.format;
+        return _.omit(schema, 'format');
```

Comment:
@webholik Why was this change needed?

Reply:
`delete` operator here deletes the `format` property in the original schema. So after the call to `resolveRefs` we cannot check what the original format is. As you said earlier we cannot rely on the value being `<binary>` as that could come from an example. 

This change ensures that we do not pass an unsupported format to json-schema-faker and ajv, at the same time not deleting the format in the original schema.